### ISSUE TITLE
Support Cassandra 4.0

### ISSUE THREAD

**Issue by tommystendahl** [Ericsson/ecaudit#187]
This would make it possible to upgrade a cluster using ecAudit to Cassandra 4.0. After upgrade its possible to switch to the built in audit logging i Cassandra or continue using ecAudit if there are features in ecAudit that are missing in Cassandra that are used.

**Comment by eperott** [comment]
Hey Tommy! :)

Perhaps this would be a reasonable opportunity to drop support for some of the older flavor-branches as well? At least 2.2 since Cassandra 2.2 is no longer supported by the community.

**Comment by tommystendahl** [comment]
Hi Pelle,

Nice to hear from you and thanks for your comments, much appreciated.

I haven't considered this but your probably right, this might be the right time to drop support for 2.2 and maybe put 3.0 in maintenance mode.

**Comment by eperott** [comment]
Sounds good. I've made some preparations for this.

I've released version 2.8.0 of ecAudit on existing favors to push out pending fixes. This will be the final release for ecaudit_c2.2 which is now archived. And ecaudit_c3.0 is now in maintenance mode, so only critical fixes go in to this branch.

I have created a new branch (release/c3.11) which is now set as the default branch on GitHub. All new features should go into this branch and be merged forward to master branch which is now heading for Cassandra 4.0 compatibility. It should be possible to rebase your WIP on this without too much issues.

I've updated the CONTRIBUTION guidelines to reflect all these changes.
